### ISSUE TITLE
Dishwasher not updating

### ISSUE THREAD

**Issue by elektrolitr** [rowysock/home-assistant-HomeWhiz#106]
Hello! 
I have a WiFi dishwasher (BDIN39640A) connected to HomeWhiz integration via the HomeWhiz Cloud

After I just connect the integration, the status is 100% correct. It may even update for several times (I wasn't able to find when)

The status often (in say 50-70% cases) updates after I use the HomeWhiz application on my phone. Again, I couldn't figure out when it does happen and when not

If I don't use the application, the status tend to stick and don't change for days. 
The log always contains the same error: 

`2023-02-10 18:41:28.001 ERROR (MainThread) [homeassistant] Error doing job: Exception in callback async_track_point_in_utc_time.<locals>.run_action(<Job HassJobT...7f6757911990>>) at /usr/src/homeassistant/homeassistant/helpers/event.py:1310
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/src/homeassistant/homeassistant/helpers/event.py", line 1325, in run_action
    hass.async_run_hass_job(job, utc_point_in_time)
  File "/usr/src/homeassistant/homeassistant/core.py", line 580, in async_run_hass_job
    hassjob.target(*args)
  File "/usr/src/homeassistant/homeassistant/helpers/event.py", line 1285, in utc_converter
    hass.async_run_hass_job(job, dt_util.as_local(utc_now))
  File "/usr/src/homeassistant/homeassistant/core.py", line 583, in async_run_hass_job
    return self.async_add_hass_job(hassjob, *args)
  File "/usr/src/homeassistant/homeassistant/core.py", line 486, in async_add_hass_job
    task = self.loop.create_task(hassjob.target(*args))
TypeError: HomewhizCloudUpdateCoordinator.refresh_connection() takes 1 positional argument but 2 were given`

**Comment by elektrolitr** [comment]
Additional logs: 
`2023-02-11 23:05:30.097 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration homewhiz which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-02-11 23:05:35.211 INFO (MainThread) [custom_components.homewhiz] Setting up entry xxxx
2023-02-11 23:05:35.211 INFO (MainThread) [custom_components.homewhiz] Setting up cloud connection
2023-02-11 23:05:35.248 DEBUG (MainThread) [custom_components.homewhiz] Sensors: ['STATE', 'DISHWASHER_PROGRAM', 'SUB_STATE', 'DISHWASHER_DELAY', 'DISHWASHER_DURATION', 'DISHWASHER_REMAINING', 'SETTING_DETERGENT_TYPE', 'DISHWASHER_RINSE_AID_LEVEL', 'DISHWASHER_LIQUID_DETERGENT_LEVEL', 'DISHWASHER_WATER_HARDNESS']
2023-02-11 23:05:35.253 DEBUG (MainThread) [custom_components.homewhiz] Binary sensors: ['REMOTE_CONTROL', 'DISHWASHER_WARNING_NO_RINSE_AID', 'DISHWASHER_WARNING_NO_SALT', 'DISHWASHER_WARNING_NO_WATER', 'DISHWASHER_WARNING_CHECK_THE_FILTER', 'DISHWASHER_WARNING_DOOR_IS_OPEN', 'DISHWASHER_WARNING_CALL_SERVICE', 'DISHWASHER_LIQUID_DETERGENT_LOW']
2023-02-11 23:05:35.955 DEBUG (MainThread) [custom_components.homewhiz] Selects: ['STATE', 'DISHWASHER_PROGRAM', 'SETTING_DETERGENT_TYPE', 'DISHWASHER_RINSE_AID_LEVEL', 'DISHWASHER_LIQUID_DETERGENT_LEVEL', 'DISHWASHER_WATER_HARDNESS']
2023-02-11 23:05:35.961 DEBUG (MainThread) [custom_components.homewhiz] ACs: []
2023-02-11 23:05:36.331 DEBUG (MainThread) [custom_components.homewhiz] Switches: ['DISHWASHER_HYGIENE_INTENSE', 'DISHWASHER_STEAMGLOSS', 'DISHWASHER_DEEPWASH', 'DISHWASHER_FAST+', 'DISHWASHER_HALFLOAD', 'SETTING_AUTO_DOOR', 'SETTING_INNER_ILLUMINATION']
2023-02-11 23:05:36.607 INFO (MainThread) [custom_components.homewhiz] Connecting to xxxx
2023-02-11 23:05:37.366 DEBUG (MainThread) [custom_components.homewhiz] Credentials expire at: 2023-02-12 00:05:36
2023-02-11 23:05:37.865 DEBUG (MainThread) [custom_components.homewhiz] Subscribe to update result: {'packet_id': 1, 'topic': '$aws/things/xxx/shadow/update/accepted', 'qos': <QoS.AT_LEAST_ONCE: 1>}
2023-02-11 23:05:37.938 DEBUG (MainThread) [custom_components.homewhiz] Subscribe to get result: {'packet_id': 2, 'topic': '$aws/things/xxx/shadow/get/accepted', 'qos': <QoS.AT_LEAST_ONCE: 1>}
2023-02-11 23:05:37.938 DEBUG (MainThread) [custom_components.homewhiz] Force read result: {'packet_id': 3}
2023-02-11 23:05:37.939 DEBUG (MainThread) [custom_components.homewhiz] Get shadow result: {'packet_id': 4}
2023-02-11 23:05:38.026 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x93\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:05:38.026 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-11 23:05:38.147 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x93\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:05:38.147 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-11 23:08:27.148 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x90\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:08:27.148 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-11 23:08:39.257 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x90\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:08:39.258 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-11 23:09:04.042 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x90\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:09:04.043 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-11 23:09:23.822 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x8f\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:09:23.822 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-11 23:09:32.614 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x8f\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:09:32.614 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-11 23:09:49.343 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x8f\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:09:49.343 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-11 23:16:36.972 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x04\x1a\x04\x1a\x81\x88\x81\x82\x80\x80\x80\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-11 23:16:36.972 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data`

The messages " (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data" happen at time I open the dishwasher in Homewhiz app.

**Comment by TechHummel** [comment]
Hi 👋 
I don't have a HomeWhiz WiFi device - so you would need to assist me with debugging 😄 

Could you change `**kwargs: Any` here https://github.com/rowysock/home-assistant-HomeWhiz/blob/main/custom_components/homewhiz/cloud.py#L141 to  `*args: Any` and check if this might already resolve the error? The HomeWhiz integration code should be in the custom_components folder of your Home Assistant installation.

I am not sure how this relates to the behaviour you described when using the app, but I think it's worth trying nevertheless. 

More detailed explanation: The `refresh_connection` callback seems to be called with time as a positional argument, whereas the code currently expects a keyword argument https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/event.py#L1285

**Comment by elektrolitr** [comment]
Hi!

Thank you for the response! I will gladly help with debugging and hopefully with something else

I made changes you suggested and the error from the first post seems fixed: 
`2023-02-12 22:26:08.000 DEBUG (MainThread) [custom_components.homewhiz] Refreshing connection
2023-02-12 22:26:08.001 INFO (MainThread) [custom_components.homewhiz] Connecting to xxxx
2023-02-12 22:26:08.503 DEBUG (MainThread) [custom_components.homewhiz] Credentials expire at: 2023-02-12 23:26:08`

Previously, it finished with an error mentioned in the first post

As to refresh of state of dishwasher, it currently (during the washing cycle happening now) refreshes once per 15-30 minutes. Previously it worked like this for some time and then stopped updates at all. I think it stopped updates when the "refresh_connection" failed. 

My goal is to create something similar to "floor light" often found on dishwashers (mine is packed with lots of features but floor light is sadly missing). Sure, I need to have reliable status. Updates once per 30 minutes are not ideal (although better than none), something between 1 and 5 minutes would be much more reasonable. 

Am I correct suggesting that events 
`2023-02-12 22:26:09.170 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x02%\x03%\x80\x80\x81\x82\x80\x80\x81\x80\x00\x01Q\x1e\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-12 22:26:09.171 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-12 22:26:09.277 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x02%\x03%\x80\x80\x81\x82\x80\x80\x81\x80\x00\x01Q\x1e\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-12 22:26:09.277 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data`
are the messages received from HomeWhiz platform? is there any means to poll HomeWhiz manually? Should be, as when I open the HomeWhiz app, it updates the data at once. 

![image](https://user-images.githubusercontent.com/121390684/218336238-03da4f4d-1e3f-422d-90eb-77d6db981a5d.png)

**Comment by TechHummel** [comment]
That's great!

I think the integration is actually subscribing to updates on HomeWhiz side. The logs you showed are messages received, yes 👍 

Could you try the following code adjustments (same file):

Line 5:
`from datetime import datetime, timedelta`

Line 11:
`from homeassistant.helpers.event import async_track_point_in_time, async_track_time_interval`

And then, between line 127 and 128, add:
```py
        async_track_time_interval(
            hass=self.hass,
            action=self.force_read,
            interval=timedelta(minutes=5)
        )
        async_track_time_interval(
            hass=self.hass,
            action=self.get_shadow,
            interval=timedelta(minutes=5)
        )
```

**Comment by elektrolitr** [comment]
Thanks! 
I've updated the code as per your instructions and there is activity each 5 minutes, but ends with error: 

`2023-02-13 08:12:20.176 ERROR (SyncWorker_6) [homeassistant] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
TypeError: HomewhizCloudUpdateCoordinator.get_shadow() takes 1 positional argument but 2 were given
2023-02-13 08:12:20.179 ERROR (SyncWorker_3) [homeassistant] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
TypeError: HomewhizCloudUpdateCoordinator.force_read() takes 1 positional argument but 2 were given`

Trying to fix this, I changed the line 173 from: 
`   def get_shadow(self) -> None:`
to 
`   def get_shadow(self, *args: Any) -> None:`
putting a dummy argument in the definition
probably, there's a better solution, but it somewhat worked and the error was now in another method:
`Traceback (most recent call last):
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
TypeError: HomewhizCloudUpdateCoordinator.force_read() takes 1 positional argument but 2 were given
2023-02-13 08:23:34.300 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x03\x80\x80\x80\x80\x00\x08\x00\x00\x01\x03%\x00\x00\x01\x02\x00\x00\x00\x00\x00\x01\xff\x14 \x00\x00\x01\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-13 08:23:34.301 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data`
but, as you can see, it still got a new status message!

So yes, I updated the line 158 to: 
`   def force_read(self, *args: Any) -> None:`

Now the log looks reasonable: 
`2023-02-13 08:31:31.240 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-13 08:36:31.108 DEBUG (SyncWorker_7) [custom_components.homewhiz] Force read result: {'packet_id': 5}
2023-02-13 08:36:31.109 DEBUG (SyncWorker_4) [custom_components.homewhiz] Get shadow result: {'packet_id': 6}
2023-02-13 08:36:31.244 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x03\x80\x80\x80\x80\x00\x08\x00\x00\x01\x03%\x00\x00\x01\x02\x00\x00\x00\x00\x00\x01\xff\x14 \x00\x00\x01\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-13 08:36:31.245 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-13 08:36:31.267 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x03\x80\x80\x80\x80\x00\x08\x00\x00\x01\x03%\x00\x00\x01\x02\x00\x00\x00\x00\x00\x01\xff\x14 \x00\x00\x01\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-13 08:36:31.267 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data`

maybe except there are two messages received at the same time

I'll test the solution overnight with the washing cycle and report back

**Comment by TechHummel** [comment]
That looks promising! I'm glad you could fix the errors too 👍

But yes, in the code is a force read and a shadow read or get - I'm honestly not sure what the shadow is for, but that is the reason there are two messages.

**Comment by elektrolitr** [comment]
currently it works and the updates are coming as expected, but the number of simultaneous "Get Shadow Result" and "Force Read Result" calls is growing. 
I omitted the "message received' stuff

`2023-02-13 14:21:31.762 DEBUG (Dummy-4) [custom_components.homewhiz] Manually updated homewhiz data
2023-02-13 14:26:31.342 DEBUG (SyncWorker_4) [custom_components.homewhiz] Force read result: {'packet_id': 137}
2023-02-13 14:26:31.343 DEBUG (SyncWorker_5) [custom_components.homewhiz] Get shadow result: {'packet_id': 138}
2023-02-13 14:26:31.344 DEBUG (SyncWorker_5) [custom_components.homewhiz] Force read result: {'packet_id': 139}
2023-02-13 14:26:31.345 DEBUG (SyncWorker_5) [custom_components.homewhiz] Get shadow result: {'packet_id': 140}
2023-02-13 14:26:31.583 DEBUG (SyncWorker_3) [custom_components.homewhiz] Force read result: {'packet_id': 141}
2023-02-13 14:26:31.584 DEBUG (SyncWorker_6) [custom_components.homewhiz] Get shadow result: {'packet_id': 142}
2023-02-13 14:26:31.587 DEBUG (SyncWorker_7) [custom_components.homewhiz] Force read result: {'packet_id': 143}
2023-02-13 14:26:31.588 DEBUG (SyncWorker_6) [custom_components.homewhiz] Get shadow result: {'packet_id': 144}
2023-02-13 14:26:31.597 DEBUG (SyncWorker_0) [custom_components.homewhiz] Force read result: {'packet_id': 145}
2023-02-13 14:26:31.605 DEBUG (SyncWorker_2) [custom_components.homewhiz] Force read result: {'packet_id': 147}
2023-02-13 14:26:31.606 DEBUG (SyncWorker_2) [custom_components.homewhiz] Get shadow result: {'packet_id': 148}
2023-02-13 14:26:31.607 DEBUG (SyncWorker_4) [custom_components.homewhiz] Get shadow result: {'packet_id': 146}`

May be somehow related to "Refreshing connection" process. 
At least for several first refreshings, the number of SyncWorkers increase after refreshing the connection. 

I think it may be because refresh_connection calls .connect (creating new connection before killing an old one) and it adds new task to hass planner, so the tasks add up over time

Probably, we need to either:
    stop the (old) async_track_time_interval() when .refresh_connection() and .kill() is called, or 
    create the async_track_time_interval only once and not at each .connect()


And yes, I commented out the 
`        async_track_time_interval(
            hass=self.hass,
            action=self.get_shadow,
            interval=timedelta(minutes=5)
        )`
and everything seems to update fine.

**Comment by TechHummel** [comment]
Yes, you are right - that makes sense! I'll check how we can add this task only once 😄

**Comment by elektrolitr** [comment]
Hi! 
I added a flag and wrapped the hass timer setup in the logic to prevent repeated activations
https://github.com/rowysock/home-assistant-HomeWhiz/commit/d62d9dbf33701941402c1eb6acafe64fb022b3a5
nothing elegant at all, but seems to work fine and confirms the hypothesis
for each 5-min interval, there are the same 3 log messages even after the credentials update: 
`2023-02-13 20:20:57.665 DEBUG (SyncWorker_0) [custom_components.homewhiz] Force read result: {'packet_id': 12}
2023-02-13 20:20:57.761 DEBUG (Dummy-4) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x04\x00\x00\x00\x00\x00\xff\x00\x00\x83\x80\x80\x80\x80\x80\x88\x00\x03%\x03%\x83\xa4\x81\x82\x80\x80\x81\x80\x00\x01\x1d<\x00\x80\x00\x81\x16\xf8\xff\t\x00\x00\x00\x00')
2023-02-13 20:20:57.761 DEBUG (Dummy-4) [custom_components.homewhiz] Manually updated homewhiz data`

I'll test it for a while.

**Comment by TechHummel** [comment]
For testing, I would've done the same 😄 
But I just read through some documentation and I think there is a better way to solve this. Since you forked the project, could you maybe open a merge request and allow maintainers to also push commits?

**Comment by TechHummel** [comment]
Go for main! And don't worry, I was there too 😄

**Comment by scottbruns** [comment]
I also have a Beko Wi-Fi dishwasher. I applied the updated code and I can verify that the status changes are working now.

**Comment by TechHummel** [comment]
Thanks for your feedback, very appreciated 👍

**Comment by kkoouu** [comment]
Just applied above mentioned PR 107 and Beko Dishwasher is working perfectly :) Thanks a lot @TechHummel & @elektrolitr

**Comment by DjordjeMandic** [comment]
@elektrolitr Are you happy with BDIN39640A? I'm looking for dishwasher but it looks like BDIN39640A is older compared to BDIN38550C and BDIN38560C but they don't have wireless connectivity.

**Comment by elektrolitr** [comment]
Hello @DjordjeMandic, 

I have somewhat mixed feeling about the BDIN39640A. I chose it for a combination of Air-Dry, liquid dispenser and utensils tray. The good thing is that everything works. Washing quality is not stellar, but mostly OK, with some occasional specks and spots left on utensils. Liquid dispenser is something I really love and AirDry also helps with drying. 
However, there are some things I do not like:
* the door is very finicky. If you don't close it carefully, it will slightly pop out and the dishwasher will go to pause - > off and you'll have full set of dirty plates when you expect them to be clean
* no floor light/any indication outside. You will open the door with machine running and don't know if it's paused. This is the reason I'm using HASS+ Homewhiz, I made a status light by myself with ESPHOME. Sadly, with current issues plugin is not working again and I'm back to the inconvenience. Also, having to use such a "thick" solution of HASS + cloud + everything for simple status indication is just ridiculous
* the utensils tray is not a fool-proof. Depending on machine load and dishes placement, I sometimes have food leftovers still attached to utensils after a full wash. 
* Default "eco" mode is ridiculously long at 4+ hours (I think it's a common for all modern DWs) and almost half of the time it's drying with the door closed. I think it could easily cut the time by an hour by using AirDry more aggressively. I use "auto" mode instead (but sadly can't set it as default), it's a full hour faster (and sometime even more, when the dishes are not very soiled) and I haven't seen any notable difference in power consumption. 

As a bottom line, this DW is more of a geek thing and asks for a bit more attention that I would like to pay to the kitchen equipment. 

For the Wireless connectivity, I would immediately trade it for any kind of external indication. I don't use WiFi for anything else.
